cpp/include/cudf/utilities/bit.hpp

```diff
@@ -121,8 +104,7 @@ __device__ inline void set_bit(bitmask_type* bitmask, size_type bit_index) {
  * @param n The number of least significant bits to set
  * @return A bitmask word with `n` least significant bits set
  *---------------------------------------------------------------------------**/
-__device__ constexpr inline bitmask_type set_least_significant_bits(
-    size_type n) {
+constexpr inline bitmask_type set_least_significant_bits(size_type n) {
   assert(0 < n < detail::size_in_bits<bitmask_type>());
```

Comment:
Do we want assert in device code?

Reply:
I think it's nicer than to have a silent UB error. And since it's only present in debug builds, it can be helpful for debugging.